scripts/gen_vimdoc.py

```diff
@@ -187,6 +187,23 @@
         'module_override': {},
         'append_only': [],
     },
+    'diagnostic': {
+        'mode': 'lua',
+        'filename': 'diagnostic.txt',
+        'section_start_token': '*diagnostic-api*',
+        'section_order': [
+            'diagnostic.lua',
+        ],
+        'files': os.path.join(base_dir, 'runtime/lua/vim/diagnostic.lua'),
+        'file_patterns': '*.lua',
+        'fn_name_prefix': '',
+        'section_name': {'diagnostic.lua': 'diagnostic'},
+        'section_fmt': lambda _: 'DIAGNOSTIC API',
+        'helptag_fmt': lambda _: f'*diagnostic-api*',
+        'fn_helptag_fmt': lambda fstem, name: f'*{fstem}.{name}()*',
```

Comment:
not sure why treesitter docs don't have `vim.*` helptags
```suggestion
        'fn_helptag_fmt': lambda fstem, name: f'*vim.{fstem}.{name}()*',
```